### PR TITLE
[DEV] Added a badge on the speaker icon

### PR DESCRIPTION
**Pull request by djavan-bertrand**
This badge is added when the speaker has at least a ribbon.

### CODE REVIEW

File: app/src/main/java/fr/paug/androidmakers/ui/activity/DetailActivity.java [hunk 1]
```diff
@@ -126,6 +126,9 @@ public void onChipClick(View view) {
                 final DetailViewSpeakerInfoElementBinding speakerInfoElementBinding = DataBindingUtil.inflate(getLayoutInflater(), R.layout.detail_view_speaker_info_element, null, false);
                 speakerInfoElementBinding.speakerBio.setMovementMethod(LinkMovementMethod.getInstance());
                 speakerInfoElementBinding.setSpeaker(speaker);
+                if (speaker.ribbonList != null && !speaker.ribbonList.isEmpty()) {
```
Comment (djavan-bertrand) [inline app/src/main/java/fr/paug/androidmakers/ui/activity/DetailActivity.java]: Note to reviewer: we can add (in this PR or later) a way to sort ribbons.

**Comment by benju69** [comment]
Cool :) t'as trouvé le ruban où ?

**Comment by benju69** [comment]
ça aurait été mieux en vectoriel le badge non ?

File: app/src/main/java/fr/paug/androidmakers/ui/activity/DetailActivity.java (hunk 1 above)
Comment (n0tnull) [reply app/src/main/java/fr/paug/androidmakers/ui/activity/DetailActivity.java]: We should have a method in the speaker object to get the ribbon. If we do that, we can remove this code and directly use it directly via the speaker in the XML with databinding.

What do you think ?

File: app/src/main/java/fr/paug/androidmakers/model/Ribbon.java [hunk 2]
```diff
@@ -28,6 +28,18 @@ public int getRibbonIcon() {
                     return -1;
             }
         }
+
+        @DrawableRes
+        public int getBadgeIcon() {
+            switch (this) {
+                case GDE:
+                    return R.drawable.gde_badge;
```
Comment (n0tnull) [inline app/src/main/java/fr/paug/androidmakers/model/Ribbon.java]: Same comment as @benju69 : Why not a vector drawable ?

File: app/src/main/java/fr/paug/androidmakers/ui/activity/DetailActivity.java (hunk 1 above)
Comment (djavan-bertrand) [reply app/src/main/java/fr/paug/androidmakers/ui/activity/DetailActivity.java]: Ok, will try to do it tonight.

**Comment by djavan-bertrand** [comment]
To answer your questions:
- I made the badge on Sketch, based on the/an official badge.
- I can try to use it as a vector drawable but I'm not sure all vector features I use (mask, shadows...) are implemented by vector drawable. We'll see!
- Ok to move access to the ribbon directly in the speaker object.

**Pull request merged**